### PR TITLE
drop second shell by inlining xilinx-shell into coyote one

### PR DESCRIPTION
**Pull request by Mic92**
This makes sure that both are providing the same libc version. Also the project patch is no longer needed.

### CODE REVIEW

**Comment by jedichen121** [comment]
There's still some errors. 

For build_perf_host_hw, I got the following new error:
image


For build_perf_rdma_host_hw, it's still the same error:
image

Besides, the error 
`
/share/xilinx/Vitis/2021.2/settings64.sh: line 5: /tools/Xilinx/DocNav/.settings64-DocNav.sh: No such file or directory
` 
can be fixed by copying that file from Sakura or Momiji. I tried before and it worked.

File: ws22/fpga-os/coyote.nix [hunk 1]
```diff
@@ -1,37 +1,69 @@
-with import <nixpkgs> {}; 
+{ pkgs ? import (fetchTarball "https://github.com/NixOS/nixpkgs/archive/1e383aada51b416c6c27d4884d2e258df201bc11.tar.gz") {}
```
Comment (Mic92) [inline ws22/fpga-os/coyote.nix:1]: I pinned now the commit for reproducibility of. This definitely works on ryan:

```
nix-shell ../coyote.nix
mkdir hw/build && cd hw/build
cmake .. -DFDEV_NAME=u50 -DEXAMPLE=perf_fpga
make shell compile
```

Please reproduce the error on this machine.

jinja2 can also imported in this shell without PYTHONPATH set:

```
[joerg@ryan fpga-os]$ PYTHONPATH= python -mjinja2
/nix/store/zlj86pbv78y3vmwv4lra2n0dbssg8sh5-python3-3.10.10-env/bin/python3.10: No module named jinja2.__main__; 'jinja2' is a package and cannot be directly executed
```

Comment (Mic92) [reply ws22/fpga-os/coyote.nix:1]: also make sure you actually have this branch checkout.

Comment (jedichen121) [reply ws22/fpga-os/coyote.nix:1]: There's still the jinja2 error on ryan. I ran `PYTHONPATH= python -mjinja2` but it didn't solve the error?

Comment (jedichen121) [reply ws22/fpga-os/coyote.nix:1]: To reproduce the other error on ryan.

```
cd Coyote # make sure you are on the latest branch
mkdir build_perf_rdma_host_hw && cd build_perf_rdma_host_hw
cmake ../hw/ -DFDEV_NAME=u280 -DEXAMPLE=perf_rdma_host -DACLK_F=250
make shell

```

Comment (Mic92) [reply ws22/fpga-os/coyote.nix:1]: This is not the command you gave me.

Comment (Mic92) [reply ws22/fpga-os/coyote.nix:1]: > There's still the jinja2 error on ryan. I ran `PYTHONPATH= python -mjinja2` but it didn't solve the error?

This was too prove that python can import jinja2.

Comment (dgiantsidi) [reply ws22/fpga-os/coyote.nix:1]: Hi both, 

This is just to let you know that I tried building the code with the new updates on both ryan and amy. 
I am able to reproduce the exact same error that @jedichen121 posted before. The problem looks like to be with `-DEXAMPLE=perf_rdma_host`. The default example (perf_fpga) compiles successfully. However, we would like to have the rdma example up and running too.

Thanks,
Dimitra

Comment (dgiantsidi) [reply ws22/fpga-os/coyote.nix:1]: Hi is there any update on this?

**Pull request closed without merging**